e2e/awspcaissuer_test.go

```diff
@@ -0,0 +1,237 @@
+package main
+
+import (
+	"context"
+	"log"
+	"os"
+	"testing"
+
+	cmclientv1 "github.com/cert-manager/cert-manager/pkg/client/clientset/versioned/typed/certmanager/v1"
+	"github.com/cucumber/godog"
+
+	"github.com/aws/aws-sdk-go-v2/aws"
+	"github.com/aws/aws-sdk-go-v2/config"
+	"github.com/cert-manager/aws-privateca-issuer/pkg/api/v1beta1"
+	clientV1beta1 "github.com/cert-manager/aws-privateca-issuer/pkg/clientset/v1beta1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/tools/clientcmd"
+)
+
+var testContext *TestContext
+
+// These are variables shared by all of the tests
+type TestContext struct {
+	iclient   *clientV1beta1.Client
+	cmClient  *cmclientv1.CertmanagerV1Client
+	clientset *kubernetes.Clientset
+	xaCfg     aws.Config
+	caArns    map[string]string
+
+	region, accessKey, secretKey, endEntityResourceShareArn, subordinateCaResourceShareArn, userName, policyArn string
```

Comment:
Would having a TestContext per feature file make these objects less complex? There has been discussion about how cucumber context ends up being very heavy and confusing, and I'm wondering if we can make it more user friendly by splitting up objects.